app/scripts/modules/core/src/managed/ManagedMenuItem.tsx

```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+import { MenuItem } from 'react-bootstrap';
+import { $q } from 'ngimport';
+import { IPromise } from 'angular';
+
+import { Application } from 'core/application';
+import { IManagedResource } from 'core/domain';
+import { confirmNotManaged } from './toggleResourceManagement';
+
+interface IManagedMenuItemProps {
+  resource: IManagedResource;
+  application: Application;
+  onClick?: () => void;
+  children: React.ReactNode;
+}
+
+export const ManagedMenuItem = ({ resource, application, onClick, children }: IManagedMenuItemProps) => {
```

Comment:
not sure I'm in love with the component name, but also don't think it really matters much and couldn't think of a better one 🤷‍♂ 

carry on.

Reply:
I also had trouble with the name of this. Alternatives I disliked more than this one:
* ManagedResourceMenuItem
* ManagedResourceRestrictedMenuItem
* RestrictedActionManagedMenuItem
...

Reply:
yeah strong agree. not worth the effort.